cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -583,11 +697,23 @@ struct preprocessed_table {
     return _depths.size() ? std::optional<device_span<int const>>(_depths) : std::nullopt;
   }
 
+  // TODO: span of spans?
+  [[nodiscard]] std::optional<device_span<dremel_device_view const>> dremel_device_views() const
```

Comment:
Humnn, a span can have zero size. So maybe we don't need to have `optional` here.